src/engraving/compat/midi/midirender.h

```diff
@@ -62,76 +62,39 @@ class RangeMap
 
 class MidiRenderer
 {
-    Score* score{ nullptr };
-    bool needUpdate = true;
-    int minChunkSize = 0;
-
 public:
-    class Chunk
+    struct Context
     {
-        int _tickOffset;
-        Measure const* first;
-        Measure const* last;
-
-    public:
-        Chunk(int tickOffset, Measure const* fst, Measure const* lst)
-            : _tickOffset(tickOffset), first(fst), last(lst) {}
-
-        Chunk()     // "invalid chunk" constructor
-            : _tickOffset(0), first(nullptr), last(nullptr) {}
-
-        operator bool() const {
-            return bool(first);
-        }
-        int tickOffset() const { return _tickOffset; }
-        Measure const* startMeasure() const { return first; }
-        Measure const* endMeasure() const { return last ? last->nextMeasure() : nullptr; }
-        Measure const* lastMeasure() const { return last; }
-        int tick1() const { return first->tick().ticks(); }
-        int tick2() const { return last ? last->endTick().ticks() : tick1(); }
-        int utick1() const { return tick1() + tickOffset(); }
-        int utick2() const { return tick2() + tickOffset(); }
+        SynthesizerState synthState;
+        bool metronome{ true };
+
+        Context() {}
     };
 
-private:
-    std::vector<Chunk> chunks;
+    explicit MidiRenderer(Score* s);
 
-    void updateChunksPartition();
-    static bool canBreakChunk(const Measure* last);
-    void updateState();
+    void renderScore(EventMap* events, const Context& ctx);
 
-    void renderStaffChunk(const Chunk&, EventMap* events, const Staff* sctx, PitchWheelRenderer& pitchWheelRenderer);
+    static const int ARTICULATION_CONV_FACTOR { 100000 };
 
-    void renderSpanners(const Chunk&, EventMap* events, PitchWheelRenderer& pitchWheelRenderer);
+private:
 
-    void renderMetronome(const Chunk&, EventMap* events);
-    void renderMetronome(EventMap* events, Measure const* m, const Fraction& tickOffset);
+    Score* score{ nullptr };
```

Comment:
should be "score = nullptr" (and other places)

Reply:
fixed